### ISSUE TITLE
[MNG-910] Plugin descriptor in pom.xml causes NPE when DefaultPluginManager populates the plugin's fields.

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#3802]
**[Corridor Software Developer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=corridor)** opened **[MNG-910](https://issues.apache.org/jira/browse/MNG-910?redirect=false)** and commented

The attached pom.xml reliably causes an NullPointerException (stacktrace also attached) by attempting to describe the use of the xdoclet plugin. Issue occurs even with little or no configuration in the actual plugin.

End of Stacktrace:
java.lang.NullPointerException
at org.codehaus.plexus.component.configurator.converters.ComponentValueSetter.\<init>(ComponentValueSetter.java:65)
at org.codehaus.plexus.component.configurator.converters.composite.ObjectWithFieldsConverter.processConfiguration(ObjectWithFieldsConverter.java:112)
at org.codehaus.plexus.component.configurator.OverrideComponentConfigurator.configureComponent(OverrideComponentConfigurator.java:41)
at org.apache.maven.plugin.DefaultPluginManager.populatePluginFields(DefaultPluginManager.java:1014)
at org.apache.maven.plugin.DefaultPluginManager.getConfiguredMojo(DefaultPluginManager.java:515)
at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:330)

---

**Affects:** 2.0-beta-1

**Attachments:**
- [testCase.tar.gz](https://issues.apache.org/jira/secure/attachment/12711513/testCase.tar.gz) (_618 bytes_)
- [TestcaseNPE.txt](https://issues.apache.org/jira/secure/attachment/12711511/TestcaseNPE.txt) (_4.09 kB_)

**Comment by jira-importer** [comment]
**[Corridor Software Developer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=corridor)** commented

environment issue. please disregard.

**Comment by jira-importer** [comment]
**[Trygve Laugstøl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=trygvis@inamo.no)** commented

If this provoces a NPE in the core this is still something that has to be fixed.

**Comment by jira-importer** [comment]
**[Kenney Westerhof](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kenney)** commented

Which version of the xdoclet plugin did you use?

**Comment by jira-importer** [comment]
**[Corridor Software Developer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=corridor)** commented

outdated 1.0 snapshot, I believe it was maven-xdoclet-plugin-1.0-20050615.091250-3.jar or maven-xdoclet-plugin-1.0-20050813.152803-4.jar. Recall per our conversation earlier today [yesterday depending on tz] that I updated and the problem went away.

Trygve, fair enough. Help me out, what project is org.codehaus.plexus.component.configurator.converters.ComponentValueSetter in? I would have just submitted a patch has I resolved the artifact & svn tree.
